test/jdk/java/util/Collection/MOAT.java

```diff
@@ -887,6 +887,41 @@ private static void testStringElement(Collection<Integer> c) {
         catch (Throwable t) { unexpected(t); }
     }
 
+    private static void testAddAll(Collection<Integer> c) {
+        if (!supportsAdd(c))
+            return;
```

Comment:
There's already a check for `supportsAdd()` in the testCollection1() method, which is the only caller, so it doesn't seem necessary to have one here.